socorro/external/postgresql/dbapi2_util.py

```diff
@@ -15,6 +17,49 @@ class SQLDidNotReturnSingleRow (Exception):
     pass
 
 
+#==============================================================================
+class FetchAllSequence(Sequence):
+    """A sequence wrapper for results of a PG cursor's fetchall()
+    that when supplied with the cursor description gives you an
+    iterable that works the same as regular fetchall() but
+    also offers the convenience method zipped().
+
+    You can use this like this::
+
+        >>> sql = "select col1, col2 from some_table"
+        >>> cursor = connection.cursor()
+        >>> cursor.excute(sql, ())
+        >>> seq = FetchAllSequence(cursor.fetchall(), cursor.description)
+        >>> seq.zipped()
+        [{'col1': value1, 'col2': value2}, {'col1': valueA, 'col2': valueB}]
+
+    """
+
+    def __init__(self, rows, description):
+        self.rows = rows
+        self.description = description
+
+    def __getitem__(self, index):
+        return self.rows[index]
+
+    def __len__(self):
+        return len(self.rows)
+
+    def __contains__(self, value):
+        return value in self.rows
+
+    def __iter__(self):
+        for x in self.rows:
+            yield x
+
+    def __str__(self):
+        return str(self.rows)
+
+    def zipped(self):
```

Comment:
two wish list things here:
* ```zipped``` is not a very descriptive name telling me what it does. ```iterable_with_column_names``` or something like that tells me what I'm getting.
* let me optionally supply the mapping class that I want to use rather than just dict:
```python
   def zipped(self, mapping_class=dict):
       names = [x.name for x in self.description]
       return [mapping_class(zip(names, x)) for x in self.rows]
```
This would allow me to do this:
```python
from socorro.lib.util import DotDict
for x in execute_query_fetch_all(some_connection, 'select a, b from sometable').zipped(DotDict):
    print x.a, x.b
```